### PR TITLE
chore(ci): image registry consistency

### LINKED ISSUE #562: bug: API and cronjob using different images
These should match.  Fix in [PR 564](https://github.com/bcgov/nr-fom/pull/564).

Update: 564 rolled into https://github.com/bcgov/nr-fom/pull/568.

### LINKED ISSUE #567: feat: add image consistency
API CronJob images are being pulled from GHCR, while the rest are imported into the OpenShift image registry first.  Find a way to make them all consistent, ideally using OpenShift.  That way we can keep an import of images for disaster-type deployments without GitHub.

Resolved in https://github.com/bcgov/nr-fom/pull/568.

### PR DESCRIPTION
**Pull request by DerekRoberts**
Resolves https://github.com/bcgov/nr-fom/issues/567.

Update: Also resolves https://github.com/bcgov/nr-fom/issues/562.

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
 - [api](https://fom-18.apps.silver.devops.gov.bc.ca/api)
 - [admin](https://fom-18.apps.silver.devops.gov.bc.ca/admin)
 - [public](https://fom-18.apps.silver.devops.gov.bc.ca/public)

Once merged, code will be promoted and handed off to following workflow run.
 - [Main Merge Workflow](https://github.com/bcgov/nr-fom/actions/workflows/merge-main.yml)

### CODE REVIEW

**Comment by ianliuwk1019** [comment]
I am trying to understand how it works.
I have a question.
I see the parameter "**PROMOTE**" is set to `PROMOTE=ghcr.io/${{ github.repository }}/${{ matrix.name }}:${{ env.ZONE }}`.
And another parameter "**REGISTRY**" is set to `value: ghcr.io`
And the **ImageStream** tag is from: "${REGISTRY}/${PROMOTE}"
```
      tags:
        - name: ${ZONE}-api
          from:
            kind: DockerImage
            name: ${REGISTRY}/${PROMOTE}
```
So at the end is the Image that we want to use from **ImageStream** is: **"${REGISTRY}/${PROMOTE}"** = "**ghcr.io/ghcr.io**/${{ github.repository }}/${{ matrix.name }}:${{ env.ZONE }}" (that has double ghcr.io in the path, or maybe my understanding is wrong)?

**Comment by DerekRoberts** [comment]
@ianliuwk1019

ImageStreams are using the PROMOTE and GHCR REGISTRY vars to import using: `${REGISTRY}/${PROMOTE}`.

CronJobs are using those imported images from `image-registry.apps.silver.devops.gov.bc.ca/${OC_NAMESPACE}/${NAME}-${ZONE}-${COMPONENT}:${ZONE}-api`.

Keep the questions coming!  :)

**Comment by ianliuwk1019** [comment]
Thanks @DerekRoberts, now I get the imageStream import part.
My question is, is this correct when string get substituted for the import path ("**ghcr.io/ghcr.io**/${{ github.repository }}/${{ matrix.name }}:${{ env.ZONE }}")? Or, maybe it indeed needs **double** "**ghcr.io/**"?

**Comment by DerekRoberts** [comment]
@ianliuwk1019 Good catch!  Yes, totally unnessary and removed.  :)

**Comment by ianliuwk1019** [comment]
> @ianliuwk1019 Good catch! Yes, totally unnessary and removed. :)

Hi Derek, seem there are some "ghcr.io" left over at "PROMOTE" in merge-demo.yml and merge-main.yml.

**Comment by DerekRoberts** [comment]
@ianliuwk1019 Done!  Those were not introduced by this PR, but more than easy enough to roll in.  Does anything else come to mind?

File: .github/workflows/merge-main.yml [hunk 1]
```diff
@@ -150,7 +149,7 @@ jobs:
           overwrite: ${{ matrix.overwrite }}
           penetration_test: false
           parameters:
-            -p PROMOTE=ghcr.io/${{ github.repository }}/${{ matrix.name }}:test
+            -p PROMOTE=${{ github.repository }}/${{ matrix.name }}:test
```
Comment (ianliuwk1019) [inline .github/workflows/merge-main.yml:153]: Sorry, hope this is my last question.
So this one should be `${{ github.repository }}/${{ matrix.name }}:${{ env.ZONE }}` as well, not `:test` right?

**Comment by MCatherine1994** [comment]
Hi Derek, I wonder if we still need pr #564? Cause that's just changing when we add "prod" tag. But if both api and cron job point to the Openshift image stream, maybe we don't need the changes? It's still fine to use the latest test tag image?

File: .github/workflows/merge-main.yml (hunk 1 above)
Comment (ianliuwk1019) [reply .github/workflows/merge-main.yml:153]: Hi Derek,
Sorry, was I asking a wrong question?
So this one should be ${{ github.repository }}/${{ matrix.name }}:${{ env.ZONE }} as well, not :test right?

**Comment by DerekRoberts** [comment]
@MCatherine1994 @ianliuwk1019 The two PRs address separate issues.  This one is for image registry consistency, fixing nothing.  The other is for the API and cronjob image mismatch.

**Comment by DerekRoberts** [comment]
PR https://github.com/bcgov/nr-fom/pull/564 is getting rolled into this one.

File: .github/workflows/merge-main.yml (hunk 1 above)
Comment (DerekRoberts) [reply .github/workflows/merge-main.yml:153]: Both PRs have been rolled together.  Hopefully that will explain everything!

**Review by MCatherine1994 — APPROVED** [review]

**Pull request merged**